core/src/main/scala/kafka/utils/ZkUtils.scala

```diff
@@ -677,7 +677,7 @@ class ZkUtils(val zkClient: ZkClient,
     topics
   }
 
-  def getPartitionReassignmentZkData(partitionsToBeReassigned: Map[TopicAndPartition, Seq[Int]]): String = {
+  def formatAsReassignmentJson(partitionsToBeReassigned: Map[TopicAndPartition, Seq[Int]]): String = {
```

Comment:
This is an improvement, but still a bit weird to have it here. How about we introduce something like:

```scala
case class PartitionReassignment(partitions: Map[TopicAndPartition, Seq[Int]) {
  def toJson: String = ...
}
```

This is just an example, but it seems to me that having to use `ZkUtils` for this is a bit odd.